### PR TITLE
fix: uppercase Discovery.MDNS.enabled in default config

### PR DESCRIPTION
**Pull request by rafaelramalho19**
Fixes https://github.com/ipfs-shipyard/ipfs-desktop/issues/1631

### CODE REVIEW

**Comment by olizilla** [comment]
@lidel do you think we should have ipfs desktop check for the incorrect key in the config, and remove it where it finds it?

**Comment by lidel** [comment]
Good question. Right now we only control the defaults applied on clean install.

There is a need to have some generic way of migrating go-ipfs configs of _existing_ users for things like `ConnMgr` limits, so to make this and future changes easier, I've added `migrateConfig` function () which checks revision and applies one-time migration, if needed. 

@rafaelramalho19 @olizilla mind reviewing my changes?

**Review by lidel — APPROVED** [review]
LGTM, but because I've added `migrateConfig` would appreciate second set of eyes before merge

File: src/daemon/config.js [hunk 1]
```diff
@@ -52,11 +52,53 @@ function applyDefaults (ipfsd) {
...
+    config = readConfigFile(ipfsd)
+  } catch (err) {
+    // This is a best effort check, dont blow up here, that should happen else where.
+    logger.error(`[daemon] migrateConfig: error reading config file: ${err.message || err}`)
+    return
+  }
+
+  // Cleanup https://github.com/ipfs-shipyard/ipfs-desktop/issues/1631
+  if (config.Discovery && config.Discovery.MDNS && config.Discovery.MDNS.enabled) {
+    config.Discovery.MDNS.Enabled = config.Discovery.MDNS.Enabled || true
```
Comment (hacdias) [inline src/daemon/config.js:83]: Isn't this `true` everytime?

Comment (lidel) [reply src/daemon/config.js:83]: no, user could remove it or explicitly set it to false to disable noise in LAN

**Comment by lidel** [comment]
Merging to unblock #1638

**Pull request merged**